src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -285,31 +285,11 @@ public boolean isAnnotated(TypeMirror e) {
         return !e.getAnnotationMirrors().isEmpty();
     }
 
-    public boolean isAnnotationType(Element e) {
-        return new SimpleElementVisitor14<Boolean, Void>() {
-            @Override
-            public Boolean visitExecutable(ExecutableElement e, Void p) {
-                return visit(e.getEnclosingElement());
-            }
-
-            @Override
-            public Boolean visitUnknown(Element e, Void p) {
-                return false;
-            }
-
-            @Override
-            protected Boolean defaultAction(Element e, Void p) {
-                return e.getKind() == ANNOTATION_TYPE;
-            }
-        }.visit(e);
+    public boolean isAnnotationInterface(Element e) {
```

Comment:
Should the parameter be a `TypeElement` ?

Reply:
I guess maybe not, since it is often called with the result of `getEnclosingElement`

Reply:
> I guess maybe not, since it is often called with the result of `getEnclosingElement`

Exactly my reasoning when I asked myself that same question while changing that method.